packages/@lwc/node-reactions/src/dom-patching/node.ts

```diff
@@ -0,0 +1,198 @@
+/*
+ * Copyright (c) 2018, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: MIT
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/MIT
+ */
+
+import { getOwnPropertyDescriptor, defineProperties } from '@lwc/shared';
+import { ReactionRecord, QualifyingReactionTypes } from '../types';
+import queueReactionsForSubtree from '../core/traverse';
+import { flushQueue } from '../core/reaction-queue';
+import { isQualifyingElement, isQualifyingHost, marker } from '../core/reactions';
+
+export default function() {
+    // caching few DOM APIs
+    const { appendChild, insertBefore, removeChild, replaceChild } = Node.prototype;
+
+    const isConnectedGetter = getOwnPropertyDescriptor(Node.prototype, 'isConnected')!.get!;
+
+    defineProperties(Node.prototype, {
+        appendChild: {
+            writable: true,
+            enumerable: true,
+            configurable: true,
+            value: function(this: Node, aChild: Node) {
+                // Performance optimization: Only check if the node being added is a registered node.
+                // We made this decision because appendChild is on the critical path and cannot
+                // subject every node to the expensive check that isQualifyingElement() performs
+                if (!isQualifyingHost(aChild)) {
+                    // invoke with apply to preserve native behavior of missing arguments, invalid types etc
+                    return appendChild.call(this, aChild);
```

Comment:
Not sure if we're not doing the following for performance reasons. If so, we should update the comment.
```suggestion
                    return appendChild.apply(this, ArraySlice.call(arguments) as [Node]);
```

Reply:
Yeah, we are not doing that for performance reasons. It caused a 5% perf regression.

Man, you caught the one place that I forgot to remove the comment :)